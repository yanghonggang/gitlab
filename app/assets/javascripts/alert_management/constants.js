import { s__ } from '~/locale';

export const ALERTS_SEVERITY_LABELS = {
  CRITICAL: s__('AlertManagement|Critical'),
  HIGH: s__('AlertManagement|High'),
  MEDIUM: s__('AlertManagement|Medium'),
  LOW: s__('AlertManagement|Low'),
  INFO: s__('AlertManagement|Info'),
  UNKNOWN: s__('AlertManagement|Unknown'),
};

export const ALERTS_STATUS_TABS = [
  {
    title: s__('AlertManagement|Open'),
    status: 'OPEN',
    filters: ['TRIGGERED', 'ACKNOWLEDGED'],
  },
  {
    title: s__('AlertManagement|Triggered'),
    status: 'TRIGGERED',
    filters: 'TRIGGERED',
  },
  {
    title: s__('AlertManagement|Acknowledged'),
    status: 'ACKNOWLEDGED',
    filters: 'ACKNOWLEDGED',
  },
  {
    title: s__('AlertManagement|Resolved'),
    status: 'RESOLVED',
    filters: 'RESOLVED',
  },
  {
    title: s__('AlertManagement|All alerts'),
    status: 'ALL',
    filters: ['TRIGGERED', 'ACKNOWLEDGED', 'RESOLVED'],
  },
];

/* eslint-disable @gitlab/require-i18n-strings */

/**
 * Tracks snowplow event when user views alerts list
 */
export const trackAlertListViewsOptions = {
  category: 'Alert Management',
  action: 'view_alerts_list',
};

/**
 * Tracks snowplow event when user views alert details
 */
export const trackAlertsDetailsViewsOptions = {
  category: 'Alert Management',
  action: 'view_alert_details',
};

/**
 * Tracks snowplow event when alert status is updated
 */
export const trackAlertStatusUpdateOptions = {
  category: 'Alert Management',
  action: 'update_alert_status',
  label: 'Status',
};
