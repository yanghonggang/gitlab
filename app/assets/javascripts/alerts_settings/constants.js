import { s__ } from '~/locale';

// TODO: Remove this as part of the form old removal
export const i18n = {
  usageSection: s__(
    'AlertSettings|You must provide this URL and authorization key to authorize an external service to send alerts to GitLab. You can provide this URL and key to multiple services. After configuring an external service, alerts from your service will display on the GitLab %{linkStart}Alerts%{linkEnd} page.',
  ),
  setupSection: s__(
    "AlertSettings|Review your external service's documentation to learn where to provide this information to your external service, and the %{linkStart}GitLab documentation%{linkEnd} to learn more about configuring your endpoint.",
  ),
  errorMsg: s__('AlertSettings|There was an error updating the alert settings.'),
  errorKeyMsg: s__(
    'AlertSettings|There was an error while trying to reset the key. Please refresh the page to try again.',
  ),
  restKeyInfo: s__(
    'AlertSettings|Resetting the authorization key for this project will require updating the authorization key in every alert source it is enabled in.',
  ),
  changesSaved: s__('AlertSettings|Your integration was successfully updated.'),
  prometheusInfo: s__('AlertSettings|Add URL and auth key to your Prometheus config file'),
  integrationsInfo: s__(
    'AlertSettings|Learn more about our our upcoming %{linkStart}integrations%{linkEnd}',
  ),
  resetKey: s__('AlertSettings|Reset key'),
  copyToClipboard: s__('AlertSettings|Copy'),
  apiBaseUrlLabel: s__('AlertSettings|API URL'),
  authKeyLabel: s__('AlertSettings|Authorization key'),
  urlLabel: s__('AlertSettings|Webhook URL'),
  activeLabel: s__('AlertSettings|Active'),
  apiBaseUrlHelpText: s__('AlertSettings|URL cannot be blank and must start with http or https'),
  testAlertInfo: s__('AlertSettings|Test alert payload'),
  alertJson: s__('AlertSettings|Alert test payload'),
  alertJsonPlaceholder: s__('AlertSettings|Enter test alert JSON....'),
  testAlertFailed: s__('AlertSettings|Test failed. Do you still want to save your changes anyway?'),
  testAlertSuccess: s__(
    'AlertSettings|Test alert sent successfully. If you have made other changes, please save them now.',
  ),
  authKeyRest: s__(
    'AlertSettings|Authorization key has been successfully reset. Please save your changes now.',
  ),
  integration: s__('AlertSettings|Integration'),
};

// TODO: Delete as part of old form removal in 13.6
export const integrationTypes = [
  { value: 'HTTP', text: s__('AlertSettings|HTTP Endpoint') },
  { value: 'PROMETHEUS', text: s__('AlertSettings|External Prometheus') },
  { value: 'OPSGENIE', text: s__('AlertSettings|Opsgenie') },
];

export const integrationTypesNew = [
  { value: '', text: s__('AlertSettings|Select integration type') },
  ...integrationTypes,
];

export const typeSet = {
  http: 'HTTP',
  prometheus: 'PROMETHEUS',
  opsgenie: 'OPSGENIE',
};

export const integrationToDeleteDefault = { id: null, name: '' };

export const JSON_VALIDATE_DELAY = 250;

export const targetPrometheusUrlPlaceholder = 'http://prometheus.example.com/';
export const targetOpsgenieUrlPlaceholder = 'https://app.opsgenie.com/alert/list/';

export const sectionHash = 'js-alert-management-settings';

/* eslint-disable @gitlab/require-i18n-strings */

/**
 * Tracks snowplow event when user views alerts integration list
 */
export const trackAlertIntegrationsViewsOptions = {
  category: 'Alert Integrations',
  action: 'view_alert_integrations_list',
};
