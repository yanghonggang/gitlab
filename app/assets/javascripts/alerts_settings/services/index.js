/* eslint-disable @gitlab/require-i18n-strings */
import axios from '~/lib/utils/axios_utils';

export default {
  // TODO: All this code save updateTestAlert will be deleted as part of: https://gitlab.com/gitlab-org/gitlab/-/issues/255501
  updateGenericKey({ endpoint, params }) {
    return axios.put(endpoint, params);
  },
  updatePrometheusKey({ endpoint }) {
    return axios.post(endpoint);
  },
  updateGenericActive({ endpoint, params }) {
    return axios.put(endpoint, params);
  },
  updatePrometheusActive({ endpoint, params: { token, config, url, redirect } }) {
    const data = new FormData();
    data.set('_method', 'put');
    data.set('authenticity_token', token);
    data.set('service[manual_configuration]', config);
    data.set('service[api_url]', url);
    data.set('redirect_to', redirect);

    return axios.post(endpoint, data, {
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
    });
  },
  updateTestAlert({ endpoint, data, token }) {
    return axios.post(endpoint, data, {
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
    });
  },
};
