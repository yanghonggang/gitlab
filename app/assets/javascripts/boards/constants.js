export const BoardType = {
  project: 'project',
  group: 'group',
};

export const ListType = {
  assignee: 'assignee',
  milestone: 'milestone',
  backlog: 'backlog',
  closed: 'closed',
  label: 'label',
  promotion: 'promotion',
  blank: 'blank',
};

export const inactiveId = 0;

export const ISSUABLE = 'issuable';
export const LIST = 'list';

/* eslint-disable-next-line @gitlab/require-i18n-strings */
export const DEFAULT_LABELS = ['to do', 'doing'];

export default {
  BoardType,
  ListType,
  DEFAULT_LABELS,
};
