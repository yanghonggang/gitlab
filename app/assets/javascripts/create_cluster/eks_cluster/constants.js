export const DEFAULT_REGION = 'us-east-2';

export const KUBERNETES_VERSIONS = [
  { name: '1.14', value: '1.14' },
  { name: '1.15', value: '1.15' },
  { name: '1.16', value: '1.16', default: true },
  { name: '1.17', value: '1.17' },
];
