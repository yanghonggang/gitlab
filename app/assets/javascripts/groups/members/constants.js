export const GROUP_MEMBER_BASE_PROPERTY_NAME = 'group_member';
export const GROUP_MEMBER_ACCESS_LEVEL_PROPERTY_NAME = 'access_level';

export const GROUP_LINK_BASE_PROPERTY_NAME = 'group_link';
export const GROUP_LINK_ACCESS_LEVEL_PROPERTY_NAME = 'group_access';
