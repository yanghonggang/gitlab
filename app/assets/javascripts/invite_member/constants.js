export const OPEN_MODAL = 'openModal';
export const MODAL_ID = 'invite-member-modal';
