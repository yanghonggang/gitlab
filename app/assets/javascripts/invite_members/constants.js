export const USER_SEARCH_DELAY = 200;
