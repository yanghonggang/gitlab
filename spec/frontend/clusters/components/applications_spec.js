import { shallowMount, mount } from '@vue/test-utils';
import Applications from '~/clusters/components/applications.vue';
import { CLUSTER_TYPE, PROVIDER_TYPE } from '~/clusters/constants';
import { APPLICATIONS_MOCK_STATE } from '../services/mock_data';
import eventHub from '~/clusters/event_hub';
import ApplicationRow from '~/clusters/components/application_row.vue';
import KnativeDomainEditor from '~/clusters/components/knative_domain_editor.vue';
import CrossplaneProviderStack from '~/clusters/components/crossplane_provider_stack.vue';
import IngressModsecuritySettings from '~/clusters/components/ingress_modsecurity_settings.vue';
import FluentdOutputSettings from '~/clusters/components/fluentd_output_settings.vue';

describe('Applications', () => {
  let wrapper;

  beforeEach(() => {
    gon.features = gon.features || {};
  });

  const createApp = ({ applications, type, props } = {}, isShallow) => {
    const mountMethod = isShallow ? shallowMount : mount;

    wrapper = mountMethod(Applications, {
      stubs: { ApplicationRow },
      propsData: {
        type,
        applications: { ...APPLICATIONS_MOCK_STATE, ...applications },
        ...props,
      },
    });
  };

  const createShallowApp = options => createApp(options, true);
  const findByTestId = id => wrapper.find(`[data-testid="${id}"]`);
  afterEach(() => {
    wrapper.destroy();
  });

  describe('Project cluster applications', () => {
    beforeEach(() => {
      createApp({ type: CLUSTER_TYPE.PROJECT });
    });

    it('renders a row for Ingress', () => {
      expect(wrapper.find('.js-cluster-application-row-ingress').exists()).toBe(true);
    });

    it('renders a row for Cert-Manager', () => {
      expect(wrapper.find('.js-cluster-application-row-cert_manager').exists()).toBe(true);
    });

    it('renders a row for Crossplane', () => {
      expect(wrapper.find('.js-cluster-application-row-crossplane').exists()).toBe(true);
    });

    it('renders a row for Prometheus', () => {
      expect(wrapper.find('.js-cluster-application-row-prometheus').exists()).toBe(true);
    });

    it('renders a row for GitLab Runner', () => {
      expect(wrapper.find('.js-cluster-application-row-runner').exists()).toBe(true);
    });

    it('renders a row for Jupyter', () => {
      expect(wrapper.find('.js-cluster-application-row-jupyter').exists()).toBe(true);
    });

    it('renders a row for Knative', () => {
      expect(wrapper.find('.js-cluster-application-row-knative').exists()).toBe(true);
    });

    it('renders a row for Elastic Stack', () => {
      expect(wrapper.find('.js-cluster-application-row-elastic_stack').exists()).toBe(true);
    });

    it('renders a row for Fluentd', () => {
      expect(wrapper.find('.js-cluster-application-row-fluentd').exists()).toBe(true);
    });
    it('renders a row for Cilium', () => {
      expect(wrapper.find('.js-cluster-application-row-cilium').exists()).toBe(true);
    });
  });

  describe('Group cluster applications', () => {
    beforeEach(() => {
      createApp({ type: CLUSTER_TYPE.GROUP });
    });

    it('renders a row for Ingress', () => {
      expect(wrapper.find('.js-cluster-application-row-ingress').exists()).toBe(true);
    });

    it('renders a row for Cert-Manager', () => {
      expect(wrapper.find('.js-cluster-application-row-cert_manager').exists()).toBe(true);
    });

    it('renders a row for Crossplane', () => {
      expect(wrapper.find('.js-cluster-application-row-crossplane').exists()).toBe(true);
    });

    it('renders a row for Prometheus', () => {
      expect(wrapper.find('.js-cluster-application-row-prometheus').exists()).toBe(true);
    });

    it('renders a row for GitLab Runner', () => {
      expect(wrapper.find('.js-cluster-application-row-runner').exists()).toBe(true);
    });

    it('renders a row for Jupyter', () => {
      expect(wrapper.find('.js-cluster-application-row-jupyter').exists()).toBe(true);
    });

    it('renders a row for Knative', () => {
      expect(wrapper.find('.js-cluster-application-row-knative').exists()).toBe(true);
    });

    it('renders a row for Elastic Stack', () => {
      expect(wrapper.find('.js-cluster-application-row-elastic_stack').exists()).toBe(true);
    });

    it('renders a row for Fluentd', () => {
      expect(wrapper.find('.js-cluster-application-row-fluentd').exists()).toBe(true);
    });

    it('renders a row for Cilium', () => {
      expect(wrapper.find('.js-cluster-application-row-cilium').exists()).toBe(true);
    });
  });

  describe('Instance cluster applications', () => {
    beforeEach(() => {
      createApp({ type: CLUSTER_TYPE.INSTANCE });
    });

    it('renders a row for Ingress', () => {
      expect(wrapper.find('.js-cluster-application-row-ingress').exists()).toBe(true);
    });

    it('renders a row for Cert-Manager', () => {
      expect(wrapper.find('.js-cluster-application-row-cert_manager').exists()).toBe(true);
    });

    it('renders a row for Crossplane', () => {
      expect(wrapper.find('.js-cluster-application-row-crossplane').exists()).toBe(true);
    });

    it('renders a row for Prometheus', () => {
      expect(wrapper.find('.js-cluster-application-row-prometheus').exists()).toBe(true);
    });

    it('renders a row for GitLab Runner', () => {
      expect(wrapper.find('.js-cluster-application-row-runner').exists()).toBe(true);
    });

    it('renders a row for Jupyter', () => {
      expect(wrapper.find('.js-cluster-application-row-jupyter').exists()).toBe(true);
    });

    it('renders a row for Knative', () => {
      expect(wrapper.find('.js-cluster-application-row-knative').exists()).toBe(true);
    });

    it('renders a row for Elastic Stack', () => {
      expect(wrapper.find('.js-cluster-application-row-elastic_stack').exists()).toBe(true);
    });

    it('renders a row for Fluentd', () => {
      expect(wrapper.find('.js-cluster-application-row-fluentd').exists()).toBe(true);
    });

    it('renders a row for Cilium', () => {
      expect(wrapper.find('.js-cluster-application-row-cilium').exists()).toBe(true);
    });
  });

  describe('Helm application', () => {
    it('does not render a row for Helm Tiller', () => {
      createApp();
      expect(wrapper.find('.js-cluster-application-row-helm').exists()).toBe(false);
    });
  });

  describe('Ingress application', () => {
    it('shows the correct warning message', () => {
      createApp();
      expect(findByTestId('ingressCostWarning').element).toMatchSnapshot();
    });

    describe('with nested component', () => {
      const propsData = {
        applications: {
          ingress: {
            title: 'Ingress',
            status: 'installed',
          },
        },
      };

      beforeEach(() => createShallowApp(propsData));

      it('renders IngressModsecuritySettings', () => {
        const modsecuritySettings = wrapper.find(IngressModsecuritySettings);
        expect(modsecuritySettings.exists()).toBe(true);
      });
    });

    describe('when installed', () => {
      describe('with ip address', () => {
        it('renders ip address with a clipboard button', () => {
          createApp({
            applications: {
              ingress: {
                title: 'Ingress',
                status: 'installed',
                externalIp: '0.0.0.0',
              },
            },
          });

          expect(wrapper.find('.js-endpoint').element.value).toEqual('0.0.0.0');
          expect(wrapper.find('.js-clipboard-btn').attributes('data-clipboard-text')).toEqual(
            '0.0.0.0',
          );
        });
      });

      describe('with hostname', () => {
        it('renders hostname with a clipboard button', () => {
          createApp({
            applications: {
              ingress: {
                title: 'Ingress',
                status: 'installed',
                externalHostname: 'localhost.localdomain',
                modsecurity_enabled: false,
              },
              cert_manager: { title: 'Cert-Manager' },
              crossplane: { title: 'Crossplane', stack: '' },
              runner: { title: 'GitLab Runner' },
              prometheus: { title: 'Prometheus' },
              jupyter: { title: 'JupyterHub', hostname: '' },
              knative: { title: 'Knative', hostname: '' },
              elastic_stack: { title: 'Elastic Stack' },
              fluentd: { title: 'Fluentd' },
              cilium: { title: 'GitLab Container Network Policies' },
            },
          });

          expect(wrapper.find('.js-endpoint').element.value).toEqual('localhost.localdomain');

          expect(wrapper.find('.js-clipboard-btn').attributes('data-clipboard-text')).toEqual(
            'localhost.localdomain',
          );
        });
      });

      describe('without ip address', () => {
        it('renders an input text with a loading icon and an alert text', () => {
          createApp({
            applications: {
              ingress: {
                title: 'Ingress',
                status: 'installed',
              },
            },
          });

          expect(wrapper.find('.js-ingress-ip-loading-icon').exists()).toBe(true);
          expect(wrapper.find('.js-no-endpoint-message').exists()).toBe(true);
        });
      });
    });

    describe('before installing', () => {
      it('does not render the IP address', () => {
        createApp();

        expect(wrapper.text()).not.toContain('Ingress IP Address');
        expect(wrapper.find('.js-endpoint').exists()).toBe(false);
      });
    });
  });

  describe('Cert-Manager application', () => {
    it('shows the correct description', () => {
      createApp();
      expect(findByTestId('certManagerDescription').element).toMatchSnapshot();
    });

    describe('when not installed', () => {
      it('renders email & allows editing', () => {
        createApp({
          applications: {
            cert_manager: {
              title: 'Cert-Manager',
              email: 'before@example.com',
              status: 'installable',
            },
          },
        });

        expect(wrapper.find('.js-email').element.value).toEqual('before@example.com');
        expect(wrapper.find('.js-email').attributes('readonly')).toBe(undefined);
      });
    });

    describe('when installed', () => {
      it('renders email in readonly', () => {
        createApp({
          applications: {
            cert_manager: {
              title: 'Cert-Manager',
              email: 'after@example.com',
              status: 'installed',
            },
          },
        });

        expect(wrapper.find('.js-email').element.value).toEqual('after@example.com');
        expect(wrapper.find('.js-email').attributes('readonly')).toEqual('readonly');
      });
    });
  });

  describe('Jupyter application', () => {
    describe('with ingress installed with ip & jupyter installable', () => {
      it('renders hostname active input', () => {
        createApp({
          applications: {
            ingress: {
              title: 'Ingress',
              status: 'installed',
              externalIp: '1.1.1.1',
            },
          },
        });

        expect(
          wrapper.find('.js-cluster-application-row-jupyter .js-hostname').attributes('readonly'),
        ).toEqual(undefined);
      });
    });

    describe('with ingress installed without external ip', () => {
      it('does not render hostname input', () => {
        createApp({
          applications: {
            ingress: { title: 'Ingress', status: 'installed' },
          },
        });

        expect(wrapper.find('.js-cluster-application-row-jupyter .js-hostname').exists()).toBe(
          false,
        );
      });
    });

    describe('with ingress & jupyter installed', () => {
      it('renders readonly input', () => {
        createApp({
          applications: {
            ingress: {
              title: 'Ingress',
              status: 'installed',
              externalIp: '1.1.1.1',
            },
            jupyter: { title: 'JupyterHub', status: 'installed', hostname: '' },
          },
        });

        expect(
          wrapper.find('.js-cluster-application-row-jupyter .js-hostname').attributes('readonly'),
        ).toEqual('readonly');
      });
    });

    describe('without ingress installed', () => {
      beforeEach(() => {
        createApp();
      });

      it('does not render input', () => {
        expect(wrapper.find('.js-cluster-application-row-jupyter .js-hostname').exists()).toBe(
          false,
        );
      });
    });
  });

  describe('Prometheus application', () => {
    it('shows the correct description', () => {
      createApp();
      expect(findByTestId('prometheusDescription').element).toMatchSnapshot();
    });
  });

  describe('Knative application', () => {
    const availableDomain = {
      id: 4,
      domain: 'newhostname.com',
    };
    const propsData = {
      applications: {
        knative: {
          title: 'Knative',
          hostname: 'example.com',
          status: 'installed',
          externalIp: '1.1.1.1',
          installed: true,
          availableDomains: [availableDomain],
          pagesDomain: null,
        },
      },
    };
    let knativeDomainEditor;

    beforeEach(() => {
      createShallowApp(propsData);
      jest.spyOn(eventHub, '$emit');

      knativeDomainEditor = wrapper.find(KnativeDomainEditor);
    });

    it('shows the correct description', async () => {
      createApp();
      wrapper.setProps({
        providerType: PROVIDER_TYPE.GCP,
        preInstalledKnative: true,
      });

      await wrapper.vm.$nextTick();

      expect(findByTestId('installed-via').element).toMatchSnapshot();
    });

    it('emits saveKnativeDomain event when knative domain editor emits save event', () => {
      propsData.applications.knative.hostname = availableDomain.domain;
      propsData.applications.knative.pagesDomain = availableDomain;
      knativeDomainEditor.vm.$emit('save');

      expect(eventHub.$emit).toHaveBeenCalledWith('saveKnativeDomain', {
        id: 'knative',
        params: {
          hostname: availableDomain.domain,
          pages_domain_id: availableDomain.id,
        },
      });
    });

    it('emits saveKnativeDomain event when knative domain editor emits save event with custom domain', () => {
      const newHostName = 'someothernewhostname.com';
      propsData.applications.knative.hostname = newHostName;
      propsData.applications.knative.pagesDomain = null;
      knativeDomainEditor.vm.$emit('save');

      expect(eventHub.$emit).toHaveBeenCalledWith('saveKnativeDomain', {
        id: 'knative',
        params: {
          hostname: newHostName,
          pages_domain_id: undefined,
        },
      });
    });

    it('emits setKnativeHostname event when knative domain editor emits change event', () => {
      wrapper.find(KnativeDomainEditor).vm.$emit('set', {
        domain: availableDomain.domain,
        domainId: availableDomain.id,
      });

      expect(eventHub.$emit).toHaveBeenCalledWith('setKnativeDomain', {
        id: 'knative',
        domain: availableDomain.domain,
        domainId: availableDomain.id,
      });
    });
  });

  describe('Crossplane application', () => {
    const propsData = {
      applications: {
        crossplane: {
          title: 'Crossplane',
          stack: {
            code: '',
          },
        },
      },
    };

    beforeEach(() => createShallowApp(propsData));

    it('renders the correct Component', () => {
      const crossplane = wrapper.find(CrossplaneProviderStack);
      expect(crossplane.exists()).toBe(true);
    });

    it('shows the correct description', () => {
      createApp();
      expect(findByTestId('crossplaneDescription').element).toMatchSnapshot();
    });
  });

  describe('Elastic Stack application', () => {
    describe('with elastic stack installable', () => {
      it('renders the install button enabled', () => {
        createApp();

        expect(
          wrapper
            .find(
              '.js-cluster-application-row-elastic_stack .js-cluster-application-install-button',
            )
            .attributes('disabled'),
        ).toBeUndefined();
      });
    });

    describe('elastic stack installed', () => {
      it('renders uninstall button', () => {
        createApp({
          applications: {
            elastic_stack: { title: 'Elastic Stack', status: 'installed' },
          },
        });

        expect(
          wrapper
            .find(
              '.js-cluster-application-row-elastic_stack .js-cluster-application-install-button',
            )
            .attributes('disabled'),
        ).toEqual('disabled');
      });
    });
  });

  describe('Fluentd application', () => {
    beforeEach(() => createShallowApp());

    it('renders the correct Component', () => {
      expect(wrapper.find(FluentdOutputSettings).exists()).toBe(true);
    });
  });

  describe('Cilium application', () => {
    it('shows the correct description', () => {
      createApp({ props: { ciliumHelpPath: 'cilium-help-path' } });
      expect(findByTestId('ciliumDescription').element).toMatchSnapshot();
    });
  });
});
