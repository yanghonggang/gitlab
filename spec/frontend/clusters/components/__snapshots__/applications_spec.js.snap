// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Applications Cert-Manager application shows the correct description 1`] = `
<p
  data-testid="certManagerDescription"
>
  Cert-Manager is a native Kubernetes certificate management controller that helps with issuing certificates. Installing Cert-Manager on your cluster will issue a certificate by 
  <a
    class="gl-link"
    href="https://letsencrypt.org/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Let's Encrypt
  </a>
   and ensure that certificates are valid and up-to-date.
</p>
`;

exports[`Applications Cilium application shows the correct description 1`] = `
<p
  data-testid="ciliumDescription"
>
  Protect your clusters with GitLab Container Network Policies by enforcing how pods communicate with each other and other network endpoints. 
  <a
    class="gl-link"
    href="cilium-help-path"
    rel="noopener"
    target="_blank"
  >
    Learn more about configuring Network Policies here.
  </a>
</p>
`;

exports[`Applications Crossplane application shows the correct description 1`] = `
<p
  data-testid="crossplaneDescription"
>
  Crossplane enables declarative provisioning of managed services from your cloud of choice using 
  <code>
    kubectl
  </code>
   or 
  <a
    class="gl-link"
    href="https://docs.gitlab.com/ee/user/clusters/applications.html#crossplane"
    rel="noopener noreferrer"
    target="_blank"
  >
    GitLab Integration
  </a>
  . Crossplane runs inside your Kubernetes cluster and supports secure connectivity and secrets management between app containers and the cloud services they depend on.
</p>
`;

exports[`Applications Ingress application shows the correct warning message 1`] = `
<strong
  data-testid="ingressCostWarning"
>
  Installing Ingress may incur additional costs. Learn more about 
  <a
    class="gl-link"
    href="https://cloud.google.com/compute/pricing#lb"
    rel="noopener noreferrer"
    target="_blank"
  >
    pricing
  </a>
  .
</strong>
`;

exports[`Applications Knative application shows the correct description 1`] = `
<span
  data-testid="installedVia"
>
  installed via 
  <a
    class="gl-link"
    href=""
    rel="noopener"
    target="_blank"
  >
    Cloud Run
  </a>
</span>
`;

exports[`Applications Prometheus application shows the correct description 1`] = `
<span
  data-testid="prometheusDescription"
>
  Prometheus is an open-source monitoring system with 
  <a
    class="gl-link"
    href="https://docs.gitlab.com/ee/user/project/integrations/prometheus.html"
    rel="noopener noreferrer"
    target="_blank"
  >
    GitLab Integration
  </a>
   to monitor deployed applications.
</span>
`;
